src/daft-core/src/array/ops/date.rs

```diff
@@ -195,4 +219,93 @@ impl TimestampArray {
             Box::new(date_arrow),
         )
     }
+
+    pub fn truncate(&self, interval: &str, start_time: &Option<i64>) -> DaftResult<Self> {
+        let physical = self.physical.as_arrow();
+        let DataType::Timestamp(timeunit, tz) = self.data_type() else {
+            unreachable!("Timestamp array must have Timestamp datatype")
+        };
+        let tu = timeunit.to_arrow();
+        let duration = process_interval(interval)?;
+
+        macro_rules! truncate_dt {
```

Comment:
Why do we need a macro here? Would generics suffice?

Reply:
I looked through https://docs.rs/chrono/latest/chrono/index.html#traits and couldn't find a trait to cover the `.timestamp()`, `.timestamp_millis()` etc methods. Not sure if there's a better solution other than a macro, or making a new trait for these methods?